internal/testing/replay/replay.go

```diff
@@ -33,61 +34,78 @@ import (
 	"google.golang.org/grpc"
 )
 
-// NewAWSRecorder returns a go-vcr.Recorder which reads or writes golden files from the given path.
-// The recorder uses a matching algorithm specialized to AWS HTTP API calls.
-// The done() function flushes the recorder and scrubs any sensitive data from the saved golden file.
+// Provider represents a cloud provider that we're going to record/replay
+// HTTP requests with.
+type Provider int
```

Comment:
Yes. But something has to tell the code which of them to use. I think it's cleaner to have a single function that takes a parameter to control that, than to have separate functions.

As part of issue #301, wherein we'll do less scrubbing, I hope to get rid of most of the provider-specific scrubbing. I hope it will be simple enough that NewRecorder can just take a parameter that has a list of regexes to scrub, or something similar (so then this enum would go away).